### ISSUE TITLE
Rule Proposal: func-param-newline

### ISSUE THREAD

**Issue by soda0289** [eslint/eslint#8452]
This rule was originally proposed by @jesstelford here: https://github.com/eslint/eslint/issues/6074#issuecomment-262388754 I looked through previous issues and did not see any other similar proposals.

**Please describe what the rule should do:**
This rule would enforce that function parameters or argument each appear on a new line, either always or when separated on multiple lines.

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
When set to `always-multiline`
```js
testA('foo', {
  bar: 2,
  baz: 3
}, {error: 'Invalid'});

function testB(foo, bar
baz) { };

testC(
  'string1', 'string2',
  1, 2
);

```
When set to `always`
```js
testA('foo', 'bar', baz);

function testB(foo, bar) {}

testC(
  'string1', 'string2',
  1, 2
);
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It is very common to have to break up long function argument list on to multiple lines. Either because of max line length rules or for increased readability. This rule would enforce a consistent style in which this should be done. Code can be hard to read when function arguments are not placed on separate lines and when a function takes in multiple JSON objects it can be hard to tell them apart.

This rule would also complement the code style that our current rules already enforce. Such as `object-property-newline` and `object-curly-newline`. As well as another proposed rule `function-paren-newline` https://github.com/eslint/eslint/pull/8102

**Comment by soda0289** [comment]
Is this case covered by https://github.com/eslint/eslint/pull/8102?

**Comment by soda0289** [comment]
I think is cover. I thought it only handled parenthesis but it covers everything inside of them. Sorry for the noise.
